lib/customer.rb

```diff
@@ -0,0 +1,40 @@
+require "csv"
+
+class Customer
+  attr_accessor :email, :address
+  attr_reader :id
+  
+  def initialize(id, email, address)
+    @id = id.to_i
+    @email = email
+    @address = address 
+  end
+  
+  def self.all
+    all_customers = []
+    
+    CSV.foreach("data/customers.csv") do |row|    
+      all_customers << Customer.new(row[0], row[1], {street: row[2], city: row[3], state: row[4], zip: row[5]})
+    end
+    
+    return all_customers
+  end
+  
+  def self.find(id)
+    customer = Customer.all.find do |customer|
+      customer.id == id
```

Comment:
You use `customer` in a weird way here that causes shadow variables. https://en.wikipedia.org/wiki/Variable_shadowing

prevent it like this:
```suggestion
      customer_return =Customer.all.find do |customer|
        customer.id == id
      end
```